tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Kafka/KafkaHelper.cs

```diff
@@ -326,7 +328,7 @@ internal static void CloseConsumerScope(Tracer tracer)
             TMessage message)
             where TMessage : IMessage
         {
-            if (!_headersInjectionEnabled)
+            if (!_headersInjectionEnabled || ((IDuckType)message).Instance is null)
```

Comment:
nit: Can we instead add `IDuckType` as a generic constraint to the method so we can guarantee at compile-time that the message is an `IDuckType`?

Reply:
I think we can do that _as well_ (or rather just make `IMessage` derive from `IDuckType`), but that doesn't solve the general differences in nullability I was talking about 🤔

Reply:
Added `IDuckType` to `IMessage` which avoids the cast, but just to clarify, this wasn't the location of the null-ref, it was in `GetMessageSize()`